website/src/queries-logic.js

```diff
@@ -237,6 +254,121 @@ function add_fork_elements(forkdata_array, user, repo, parentDefaultBranch) {
   }
 }
 
+function update_filter_appearance() {
+  const filter = getFilterOrDefault();
+  if (filter === '') {
+    JQ_FILTER_FIELD.removeClass('is-dark');
+  } else {
+    JQ_FILTER_FIELD.addClass('is-dark');
+  }
+}
+
+function update_filter() {
+  update_filter_appearance();
+  updateFilterFunction();
+  update_table_trying_use_filter();
+
+  updateBasedOnTable();
+}
+
+/**
+ * Rewrites the table with the specified data.
+ * @param {Array} data - Array of objects with the following keys: name, stars, forks, ahead_by, ahead_url, behind_by, behind_url, pushed_at
+ */
+function update_table(data) {
+  clearTable();
+  let table_body = getTableBody();
+  for (const currFork of data) {
+    const { name, stars, forks, ahead_by, ahead_url, behind_by, behind_url, pushed_at } = currFork;
+    const date_txt = compareDates(pushed_at, getDateCol(pushed_at));
+
+    const NEW_ROW = $('<tr>', { id: extract_username_from_fork(name), class: "useful_forks_repo" });
+    NEW_ROW.append(
+      $('<td>').html(getRepoCol(name, false)).attr("value", name),
+      $('<td>').html(UF_TABLE_SEPARATOR + getStarCol(stars)).attr("value", stars),
+      $('<td>').html(UF_TABLE_SEPARATOR + getForkCol(forks)).attr("value", forks),
+      $('<td>').html(UF_TABLE_SEPARATOR),
+      $('<td>', { class: "uf_badge" }).html(ahead_badge(ahead_by, ahead_url)).attr("value", ahead_by),
+      $('<td>').html(UF_TABLE_SEPARATOR),
+      $('<td>', { class: "uf_badge" }).html(behind_badge(behind_by, behind_url)).attr("value", behind_by),
+      $('<td>').html(UF_TABLE_SEPARATOR + date_txt).attr("value", pushed_at)
+    );
+    table_body.append(NEW_ROW);
+  }
+  sortTable();
+}
+
+/**
+ * 1. Empty filter means no filter.
+ * 2. Filter string is a list of conditions separated by spaces.
+ * 3. If a condition is invalid, it is ignored, and the rest of the conditions are applied.
+ */
+function updateFilterFunction() {
+  const filter = getFilterOrDefault();
+  if (filter === '') {
+    return; // no filter
+  }
+
+  const mapTable = {
+    'ahead': 'ahead_by',
+    'behind': 'behind_by',
+    'pushed': 'pushed_at',
+    'date': 'pushed_at',
+    'd': 'pushed_at',
+    'a': 'ahead_by',
+    'b': 'behind_by',
+    'p': 'pushed_at',
+    's': 'stars',
+    'f': 'forks',
+  };
+
+  // parse filter string into condition object
+  const conditionStrList = filter.split(' ');
+  let conditionObj = {};
+  for (const condition of conditionStrList) {
+    let [attribute, requirement] = condition.split(':');
+    if (!attribute || !requirement) {
+      continue; // invalid condition
+    }
+    const [_, operator, value] = requirement.split(/([<>=]+)/);
+    if (!operator || !value) {
+      continue; // invalid condition
+    }
+    if (attribute in mapTable) {
+      attribute = mapTable[attribute];
+    }
+    conditionObj[attribute] = { operator, value };
+  }
```

Comment:
What are your thoughts on this? Why prefer the GitHub syntax when we can support something simpler?
```suggestion
  for (const condition of conditionStrList) {
    const [attribute, operator, value] = condition.match(/(\w+)(<=|>=|[<=>])([0-9]+)/)?.slice(1);
    if (!attribute || !operator || !value) {
      continue; // invalid condition
    }
    if (attribute in mapTable) {
      attribute = mapTable[attribute];
    }
    conditionObj[attribute] = { operator, value };
  }
```

Reply:
All right. GitHub and Google chose such syntax may be because users usually search about string, and so the filters should be distinguished from that. Considering we don't need to search about any strings, a simpler syntax can be better indeed.